ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/monitor-telemetry-postcard-policy.j2

```diff
@@ -0,0 +1,65 @@
+{#
+ Copyright (c) 2023-2024 Arista Networks, Inc.
+ Use of this source code is governed by the Apache License 2.0
+ that can be found in the LICENSE file.
+#}
+{# doc - monitor telemetry postcard policy #}
+{% if monitor_telemetry_postcard_policy is arista.avd.defined %}
+
+### Monitor Telemetry Postcard Policy
+{%     if monitor_telemetry_postcard_policy.sample_policies is arista.avd.defined %}
+
+#### Sample Policy Summary
+{%         for policy in monitor_telemetry_postcard_policy.sample_policies | arista.avd.natural_sort("name") %}
+
+##### {{ policy.name }}
+{%             if policy.match_rules is arista.avd.defined %}
+
+###### Match rules
+
+| Rule Name | Rule Type | Source Prefix | Destination Prefix | Protocol | Source Ports | Destination Ports |
+| --------- | --------- | ------------- | ------------------ | -------- | ------------ | ----------------- |
+{%                 for rule in policy.match_rules | arista.avd.natural_sort("name") %}
+{%                     set protocols_list = [] %}
+{%                     set destination_ports = [] %}
+{%                     set source_ports = [] %}
+{%                     for protocol in rule.protocols | arista.avd.natural_sort("protocol") %}
+{%                         do protocols_list.append(protocol.protocol) %}
```

Comment:
It should not be required to run through all these lists here, since you use join filter below. Please revisit this.